ui: CSS modules for Table components

koorosh opened the pull request:
Depends on #47513
Related to #47527

This change refactors components to use CSS modules and
incorporate all required styles without any external
dependencies and prevent styles altering from outside.
It affects several components which tightly
coupled with StatementsTable and couldn't be changed
separately.

Following component are changed:
- HighlightedText
- Drawer
- StatementsTable
- SortableTable

Note, that `StatementsTable#makeCommonColumns` function
is refactored to provide custom styles from parent to
child components via props instead of overriding styles.

Storybook is extended to show some components as independent
units or in context of `StatementTable` component (if it is
only the way components work).

Release note: None

koorosh committed:
ui: Add Statements page storybook

To test styles isolation for statements page
we need storybook which displays entire
Statements screen only.

To make it work, RouterProvider decorator
is added which connects router to dummy (empty)
store.

`statementsPage.fixture.ts` file contains snapshot
of required props for StatementsPage component

Release note: None

koorosh committed:
ui: CSS modules for PlanView component

- refactor fonts imports to correctly resolve paths
when module is required from different locations;
- move all files related to PlanView component under
`planView` directory
- Added story for PlanView component

Release note: None

koorosh committed:
ui: CSS modules for SortableTable and StatementsTable

This change refactors components to use CSS modules and
incorporate all required styles without any external
dependencies and prevent styles altering from outside.
It affects several components which tightly
coupled with StatementsTable and couldn't be changed
separately.

Following component are changed:
- HighlightedText
- Drawer
- StatementsTable
- SortableTable

Note, that `StatementsTable#makeCommonColumns` function
is refactored to provide custom styles from parent to
child components via props instead of overriding styles.

Storybook is extended to show some components as independent
units or in context of `StatementTable` component (if it is
only the way components work).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47606)
<!-- Reviewable:end -->

koorosh committed:
ui: Refactor classNames usage with bind for PlanView component

Previously, class names were constructed by simply accessing
style modules class names and assigning it to classes. It was
cumbersome and not readable at all.

To enhance this, `classnames/bind` alternate is used, which
allows simply put class names.

Release note: None

koorosh commented:
bors r+
